doc/src/cylc-user-guide/cug.tex

```diff
@@ -6166,6 +6092,466 @@ \subsubsection{Where To Put Batch System Handler Modules}
+        'cylc_run_dir': cylc_run_dir
+    }
+    return (satisfied, results)
+\end{lstlisting}
+The \lstinline=satisified= variable is boolean (value True or False, depending
+on whether or not the trigger condition was found to be satisfied). The
+\lstinline=results= dictionary contains the names and values of all of the
+target suite state parameters. Each item in it gets qualified with the
+unique trigger label (``upstream'' here) and passed to the environment of
+dependent task jobs (the members of the \lstinline=FAM= family in this case).
+To see this, take a look at the job script for one of the downstream tasks:
+
+\begin{lstlisting}
+% cylc cat-log -f j dn f2.2011 
+...
+cylc__job__inst__user_env() {
+    # TASK RUNTIME ENVIRONMENT:
+    export upstream_suite upstream_cylc_run_dir upstream_offset \
+      upstream_message upstream_status upstream_point upstream_task
+    upstream_suite="up"
+    upstream_cylc_run_dir="/home/vagrant/cylc-run"
+    upstream_offset="None"
+    upstream_message="data ready"
+    upstream_status="succeeded"
+    upstream_point="2011"
+    upstream_task="foo"
+}
+...
+\end{lstlisting}
+
+Note that the task has to know the name (label) of the external trigger that it
+depends on - ``upstream'' in this case - in order to use this information.
+However the name could be given to the task environment in the suite
+configuration.
+
+\subsection{Custom Trigger Functions}
+\label{Custom Trigger Functions}
+
+Trigger functions are just normal Python functions, with a few special
+properties:
+\begin{myitemize}
+  \item they must be defined in a module with the same name as the function
+  \item they can be located in:
+    \begin{myitemize}
+      \item \lstinline=<cylc-dir>/lib/cylc/xtriggers/=
+      \item \lstinline=<suite-dir>/lib/python/=
+      \item (or anywhere in your Python library path)
+    \end{myitemize}
+  \item they can take arbitrary positional and keyword arguments
+  \item suite and task identity, and cycle point, can be passed to trigger
+    functions by using string templates in function arguments (see below)
+  \item integer, float, boolean, and string arguments will be recognized and
+    passed to the function as such
+  \item they must take a boolean \lstinline=debug= argument, because this will
+    be automatically supplied by the suite when running in debug mode
+  \item if a trigger function depends on files or directories (for example)
+    that might not exist when the function is first called, just return
+    unsatisified until everything required does exist.
+\end{myitemize}
+
+Note that trigger functions cannot store data Pythonically between invocations
+because each call is executed in an independent process in the process pool. If
+necessary the filesystem can be used for this purpose.
+
+The following string templates are available for use, if the trigger function
+needs any of this information, in function arguments in the suite configuration:
+\begin{myitemize}
```

Comment:
This is the explanation I was looking for earlier. It does appear to be a better way, but it will have to be another PR at this point.  I have at least done a quick fix to provide suite debug mode by string template, like the other suite and task attributes, instead of automatically adding a debug kwarg in suite debug mode (which breaks a function that does not take a debug arg).